pkg/chunk/schema.go

```diff
@@ -734,3 +774,47 @@ func (v10Entries) GetChunksForSeries(bucket Bucket, seriesID []byte) ([]IndexQue
 		},
 	}, nil
 }
+
+func (v10Entries) GetMetricsForSeries(_ Bucket, _ []byte) ([]IndexQuery, error) {
+	return nil, ErrNotSupported
+}
+
+// v11Entries builds on v10 but adds index entries for each series to store respective labels.
+type v11Entries struct {
+	v10Entries
+}
+
+func (s v11Entries) GetChunkWriteEntries(bucket Bucket, metricName string, labels labels.Labels, chunkID string) ([]IndexEntry, error) {
+	seriesID := labelsSeriesID(labels)
+	encodedThroughBytes := encodeTime(bucket.through)
+	data, err := jsoniter.ConfigFastest.Marshal(labels)
+	if err != nil {
+		return nil, err
+	}
+	entries := []IndexEntry{
+		// Entry for seriesID -> chunkID
+		{
+			TableName:  bucket.tableName,
+			HashValue:  bucket.hashKey + ":" + string(seriesID),
+			RangeValue: encodeRangeKey(encodedThroughBytes, nil, []byte(chunkID), chunkTimeRangeKeyV3),
+		},
+		// Entry for seriesID -> labels
+		{
+			TableName:  bucket.tableName,
+			HashValue:  bucket.hashKey + ":" + string(seriesID) + ":metric",
+			RangeValue: encodeRangeKey(nil, nil, nil, metricConstRangeKeyV1),
+			Value:      data,
+		},
+	}
+
+	return entries, nil
+}
+
+func (v11Entries) GetMetricsForSeries(bucket Bucket, seriesID []byte) ([]IndexQuery, error) {
+	return []IndexQuery{
+		{
+			TableName: bucket.tableName,
+			HashValue: bucket.hashKey + ":" + string(seriesID) + ":metric",
```

Comment:
Do you need to specify a range value here?

Reply:
I added a `+ ":metric"` at the end of the HasValue, so I don't think I need the RangeValue to query. (From the doc above it means I'm getting all of them)

Reply:
However the `IndexEntry` does have a RangeValue as it seems required for validation. But it is just a constant.